tests/testthat/test-trailing_blank_lines_linter.R

```diff
@@ -96,3 +98,55 @@ test_that("returns the correct linting", {
     column_number = 1L
   ), linter)
 })
+
+test_that("blank lines in chunks produce lints", {
+  linter <- trailing_blank_lines_linter()
+
+  tmp6 <- withr::local_tempfile(fileext = ".Rmd")
```

Comment:
How can I prevent `withr::local_tempfile()` from adding a terminal newline? 
Otherwise, it is not possible to swap these two ways of creating temp files for tests.

``` r
library(lintr)

tmp <- withr::local_tempfile()
cat("lm(y ~ x)", file = tmp)
lint(tmp, linters = trailing_blank_lines_linter())
#> /private/var/folders/xr/v_vddzvs33q5wg7jv9mr__4h0000gn/T/RtmpwfMgWs/file233928cbba42:1:10: style: [trailing_blank_lines_linter] Missing terminal newline.
#> lm(y ~ x)
#>          ^

tmp <- withr::local_tempfile(lines = "lm(y ~ x)")
lint(tmp, linters = trailing_blank_lines_linter())
```

<sup>Created on 2022-10-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
yea for tests where we need to exclude terminal newline, cat() is the only option, but I didn't realize that was the intent here.

Reply:
Okay, thanks for confirming. 
 
No other changes left to be made here then, at least not for the existing review comments.